js/quotechew.js

```diff
@@ -476,8 +580,50 @@ exports.quoteProcessTextBody = function quoteProcessTextBody(fullBodyText) {
         if (lastNonWhitespaceLine &&
             RE_WROTE_LINE.test(lastNonWhitespaceLine)) {
 
-          // count the newlines up to the lead-in's newline
+          // But surprise!  This could be a non-format-flowed message where the
+          // true lead-in line started on the previous line and the "wrote" bit
+          // that triggered us spills over.
+
+          // Let's start with just the line we've got...
+          // (count the newlines up to the lead-in's newline)
           var upToPoint = idxLastNonWhitespaceLineEnd;
+
+          // Now let's see if the preceding line had content.  We can infer this
+          // by whether we find a newline between the prevLast and last points.
+          // No newline means there's content on that preceding line.  (There
+          // could also be more content on the line(s) prior to that that flow
+          // into this, but that's not particularly likely unless something is
+          // going wrong with our heuristic.  At which point it's good for us
+          // to mitigate the damage.)
+          if (idxPrevLastNonWhitespaceLineEnd !== null) {
+            var considerIndex = idxPrevLastNonWhitespaceLineEnd + 1;
+            while (considerIndex < idxLastNonWhitespaceLineEnd) {
+              if (fullBodyText[considerIndex++] === '\n') {
+                break;
+              }
+            }
+
+            if (considerIndex === idxLastNonWhitespaceLineEnd) {
+              // We didn't encounter a newline.  So now we need to rewind the
```

Comment:
Given the while loop above, if considerIndex starts the while loop body as `idxLastNonWhitespaceLineEnd - 1`, but that character happens to be a \n, the loop breaks, but considerIndex is incremented to be `idxLastNonWhitespaceLineEnd`, and it would go into this block.

I am still tracing the larger logic, so perhaps `idxLastNonWhitespaceLineEnd - 1` could not ever be a \n, or not important. So just confirming that the behavior of the `considerIndex++` is desirable/safe in this case.